src/ngx_http_lua_headers.c

```diff
@@ -1002,6 +1010,14 @@ ngx_http_lua_ffi_req_set_header(ngx_http_request_t *r, const u_char *key,
         return NGX_DECLINED;
     }
 
+    for (i = key_len; i > 0; i--) {
```

Comment:
Both leading and trailing spaces must be removed

Reply:
I read the RFC document and did not find the mandatory requirement that no spaces be allowed before field-name

Reply:
https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.4

Reply:
Sorry, did you mean obs-fold?

Reply:
**A field value might be preceded and/or followed by optional
   whitespace (OWS); a single SP preceding the field-value is preferred
   for consistent readability by humans.  The field value does not
   include any leading or trailing whitespace:**

Reply:
sorry, we are talking about field name not field value

Reply:
Hmm , approving it . problem was leading leading whitespace in field name was also failing in kong